### DISCUSSION TITLE
How to setup custom queries to scan

### DISCUSSION THREAD

**Discussion by crispy-tk** [Checkmarx/kics#6192] [Q&A] [answered]
Hello! 👋 
I've been searching the oficial documentation for instructions on how to implement a private collection of custom queries but couldn't find anything (maybe I'm looking in the wrong place). Is this possible?

**Comment by cx-miguel-dasilva** [comment, marked as answer]
Hi @crispy-tk 👋,
To implement a private collection of custom queries you just need to create them following the KICS Query structure and then use the `-q` flag with the path to the folder where they are located. For reference to the query structure and the command line options, you can see the docs.
- https://docs.kics.io/latest/creating-queries/
- https://docs.kics.io/latest/commands/

**Reply by crispy-tk** [reply]
Thank you, @cxMiguelSilva !
I was unsure of using the `-q` flag as I thought it would merely change the default path to a new one.
So, will it keep looking for the default queries as well as the custom once it's set?

**Reply by cx-miguel-dasilva** [reply]
If you make use of the flag `-q` it will dismiss the default value. To make use of your custom queries alongside KICS security queries you need to specify the path to both. In case you are using docker to run KICS can do it like this:
`kics scan -q "/your/queries/path","./assets/queries" `
Note that the `./assets/queries` is the default path of the queries in the KICS docker image as referenced in the [docs](https://docs.kics.io/latest/commands/#scan_command_options).
Hope this helps! Let us know in case you have any other doubts.
